processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -1028,21 +1038,37 @@ func (c *WatchClient) getIdentifiersFromAssoc(pod *Pod) []PodIdentifier {
 		}
 
 		if !skip {
-			ids = append(ids, ret)
+			if retID4containerID != -1 {
+				// As there can be multiple container.IDs per pod,
+				// one PodIdentifier is added per container.ID.
+				cIDs := maps.Keys(pod.Containers.ByID)
+				for cID := range cIDs {
+					retCpy := ret
+					retCpy[retID4containerID] = PodIdentifierAttributeFromSource(AssociationSource{
+						From: ResourceSource,
+						Name: string(conventions.ContainerIDKey),
+					}, cID)
+					ids = append(ids, retCpy)
```

Comment:
Not sure if this will work when pod attributes are defined _after_ `container.id` given that we drop the rest here.
```
    pod_association:
      - sources:
          - from: resource_attribute
            name: container.id
          - from: resource_attribute
            name: k8s.pod.name
```

Reply:
For this case, there is the testcase called `multiple associations` where a pod attribute is defined after `container.id`.